File: frontend/src/types/recipe.ts
export interface Recipe {
  id: number
  name: string
  description: string | null
  imageUrl: string | null
  totalTime: number | null
  calories: number | null
  difficulty: string | null
  servings: number
  prepTime: number | null
  cookTime: number | null
  protein: number | null
  carbs: number | null
  fat: number | null
}

export interface Ingredient {
  id: number
  recipeId: number
  name: string
  quantity: number | null
  unit: string | null
  notes: string | null
  order: number
}

export interface Instruction {
  id: number
  recipeId: number
  stepNumber: number
  description: string
  imageUrl: string | null
}

export interface Tag {
  id: number
  recipeId: number
  tagName: string
}

export interface Allergen {
  id: number
  recipeId: number
  allergenName: string
}

export interface RecipeDetail extends Recipe {
  ingredients: Ingredient[]
  instructions: Instruction[]
  tags: Tag[]
  allergens: Allergen[]
}

export interface SearchParams {
  q?: string
  difficulty?: string
  maxTime?: number
  maxCalories?: number
  tags?: string[]
}

export interface SearchFilters {
  q: string | null
  difficulty: string | null
  maxTime: number | null
  maxCalories: number | null
  tags: string[] | null
}

export interface SearchResult {
  filters: SearchFilters
  count: number
  recipes: Recipe[]
}

File: api/MealPlanner.Api/Models/Recipe.cs
namespace MealPlanner.Api.Models;

public class Recipe
{
    public long Id { get; set; }
    public string Slug { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Difficulty { get; set; }

    public int? PrepTime { get; set; }
    public int? CookTime { get; set; }
    public int? TotalTime { get; set; }
    public int Servings { get; set; } = 2;

    public int? Calories { get; set; }
    public int? Protein { get; set; }
    public int? Carbs { get; set; }
    public int? Fat { get; set; }

    public string? ImageUrl { get; set; }
    public string? HellofreshUrl { get; set; }
    public string? PdfUrl { get; set; }
    public string Region { get; set; } = "en-US";

    public DateTime ScrapedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public ICollection<Ingredient> Ingredients { get; set; } = new List<Ingredient>();
    public ICollection<Instruction> Instructions { get; set; } = new List<Instruction>();
    public ICollection<Tag> Tags { get; set; } = new List<Tag>();
    public ICollection<Allergen> Allergens { get; set; } = new List<Allergen>();
}
